tests/rule_based_profiler/test_toolkit.py

```diff
@@ -32,6 +36,83 @@ def test_get_profiler_with_too_many_args_raises_error(
     assert "either name or ge_cloud_id" in str(e.value)
 
 
+@mock.patch("great_expectations.data_context.data_context.DataContext")
+def test_add_profiler(
+    mock_data_context: mock.MagicMock,
+    profiler_key: ConfigurationIdentifier,
+    profiler_config: RuleBasedProfilerConfig,
+):
+    mock_data_context.ge_cloud_mode.return_value = False
+    profiler = add_profiler(profiler_config, mock_data_context)
+
+    assert isinstance(profiler, RuleBasedProfiler)
+    assert profiler.name == profiler_config.name
+    assert mock_data_context.profiler_store.set.call_args == mock.call(
+        key=profiler_key, value=profiler_config
+    )
+
+
+@mock.patch("great_expectations.data_context.data_context.DataContext")
+def test_add_profiler_ge_cloud_mode(
+    mock_data_context: mock.MagicMock,
+    ge_cloud_profiler_id: str,
+    ge_cloud_profiler_key: GeCloudIdentifier,
+    profiler_config: RuleBasedProfilerConfig,
+):
+    mock_data_context.ge_cloud_mode.return_value = True
+    profiler = add_profiler(
+        profiler_config, mock_data_context, ge_cloud_id=ge_cloud_profiler_id
+    )
+
+    assert isinstance(profiler, RuleBasedProfiler)
+    assert profiler.name == profiler_config.name
+    assert mock_data_context.profiler_store.set.call_args == mock.call(
+        key=ge_cloud_profiler_key, value=profiler_config
+    )
+
+
+@mock.patch("great_expectations.data_context.data_context.DataContext")
```

Comment:
@cdkini This might need some adjustment based on the merge of #4111 -- thanks.

Reply:
I think this work as is! Let me know if that's not the case.